packages/amplify-cli/src/domain/amplify-usageData/getUsageDataUrl.ts

```diff
@@ -7,8 +7,12 @@ export const prodUrl = `https://api.cli.amplify.aws/${version}/metrics`;
  *  Usage data tracking service URL
  */
 export const getUrl = (): UrlWithStringQuery => {
-  if (process.env.AMPLIFY_CLI_BETA_USAGE_TRACKING_URL && typeof process.env.AMPLIFY_CLI_BETA_USAGE_TRACKING_URL === 'string') {
-    return url.parse(process.env.AMPLIFY_CLI_BETA_USAGE_TRACKING_URL || '');
+  if (isProduction() && !useBetaUrl()) {
+    return url.parse(prodUrl);
   }
-  return url.parse(prodUrl);
+
+  return url.parse(process.env.AMPLIFY_CLI_BETA_USAGE_TRACKING_URL || '');
```

Comment:
I'd suggest to tweak this code a bit to make sure that we `url.parse` once per process lifespan.
I.e. we cache result of this logic somewhere. I don't think we need to compute it every time we resolve `getUrl`.
Otherwise change looks good.

Reply:
updated